### PR TITLE
Fix underflow in fill_range_response_from_file function

### PR DESCRIPTION
**Pull request by eddyashton**
This pull request addresses a critical bug in the `fill_range_response_from_file` function located in `file_serving_handlers.h`. The issue involved an unchecked offset that could lead to an unsigned underflow, allowing unauthorized memory access.

### Changes made:
- **Bug Fix**: Implemented checks to prevent underflow when calculating the range for file serving. Specifically, suffix ranges are now clamped to the file size, and inclusive range ends are validated to avoid overflow.
- **Error Handling**: Added rejection for empty ranges with a `400 Bad Request` response, ensuring compliance with RFC 9110.
- **Testing**: Enhanced end-to-end tests in `e2e_operations.py` to cover the new behavior and validate the fix against various edge cases.
- **Changelog Update**: Updated `CHANGELOG.md` to reflect the changes made.

### Impact:
- The fix prevents potential memory leaks and unauthorized access to sensitive data, particularly affecting endpoints that do not require authentication. The changes ensure that the system behaves correctly under various input scenarios, enhancing overall security and stability.

### CODE REVIEW

**Review by maxtropets — APPROVED** [review]

File: src/node/rpc/file_serving_handlers.h [hunk 1]
```diff
@@ -364,6 +381,23 @@ namespace ccf::node
       }
     }
 
+    // A 206 response must describe a non-empty range in its Content-Range
+    // header, so an empty range cannot be satisfied. This catches ranges which
+    // are individually in-bounds but select no bytes, such as "bytes=-0",
+    // "bytes=50-49", or a range starting exactly at the end of the file.
+    if (range_start == range_end)
+    {
+      ctx.rpc_ctx->set_error(
```
Comment (achamayou) [inline src/node/rpc/file_serving_handlers.h:390]: @copilot this is effectively the same case and should be same error message as we return at 310, according to rfc9110 (An int-range is invalid if the last-pos value is present and less than the first-pos.). Make the error messages identical.

**Review by achamayou — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Ledger chunk metadata and snapshot scheduling are no longer restored by a transaction whose writes a concurrent view change has already discarded. Both are now updated under the same lock as the rollback, and skipped when the transaction's rollback epoch or view no longer holds (#8243).
 - A transaction whose view changed while it was committing could apply its writes to the local key-value store and then fail to replicate, leaving state that never reached consensus. The transaction's view is now validated atomically with the allocation of its version, so it is rejected before any map is modified, and `ccf::kv::CommitResult::FAIL_NO_REPLICATE` no longer implies a locally applied write (#8242).
 - A transaction in a JavaScript application endpoint which conflicts with compaction is now re-executed, rather than returning `500 Internal Server Error` (#8289).
+- A `Range` header requesting a suffix longer than the file is now clamped to the whole file, per RFC 9110. Ranges which select no bytes, such as `bytes=-0`, are now rejected with `400 Bad Request` (#8296).
```
Comment (eddyashton) [inline CHANGELOG.md]:
```suggestion
- A `Range` header requesting a suffix longer than the file is now clamped to the whole file, per RFC 9110. Ranges which select no bytes, such as `bytes=-0`, are now rejected with `400 Bad Request` (#8299).
```
